saleor/plugins/models.py

```diff
@@ -1,20 +1,25 @@
 from django.db import models
 from django.db.models import JSONField  # type: ignore
 
+from ..channel.models import Channel
 from ..core.permissions import PluginsPermissions
 from ..core.utils.json_serializer import CustomJsonEncoder
 
 
 class PluginConfiguration(models.Model):
-    identifier = models.CharField(max_length=128, unique=True)
+    identifier = models.CharField(max_length=128)
     name = models.CharField(max_length=128)
+    channel = models.ForeignKey(
+        Channel, blank=True, null=True, on_delete=models.SET_NULL
```

Comment:
I think we should remove `PluginConfiguration` assigned to the removed channel. 
```suggestion
        Channel, blank=True, null=True, on_delete=models.CASCADE
```